pkg/devfile/image/docker_compatible.go

```diff
@@ -57,21 +70,30 @@ func (o *DockerCompatibleBackend) Build(image *devfile.ImageComponent, devfilePa
 	return nil
 }
 
-func getShellCommand(cmdName string, image *devfile.ImageComponent, devfilePath string) string {
+//getShellCommand creates the docker compatible build command from detected backend,
+//container image and devfile path
+func getShellCommand(cmdName string, image *devfile.ImageComponent, devfilePath string) []string {
+	var shellCmd []string
 	imageName := image.ImageName
 	dockerfile := filepath.Join(devfilePath, image.Dockerfile.Uri)
 	buildpath := image.Dockerfile.BuildContext
 	if buildpath == "" {
 		buildpath = devfilePath
 	}
 	args := image.Dockerfile.Args
-
-	shell := fmt.Sprintf(`%s build -t "%s" -f "%s" "%s"`, cmdName, imageName, dockerfile, buildpath)
+	shellCmd = []string{
+		cmdName,
+		"build",
+		"-t",
+		imageName,
+		"-f",
+		dockerfile,
+		buildpath,
+	}
 	if len(args) > 0 {
-		shell = shell + " " + strings.Join(args, " ")
+		shellCmd = append(shellCmd, args...)
```

Comment:
Per the Devfile spec, `args` are `arguments to supply to the dockerfile build`. Since they are passed to the `<docker|podman> build ...` command, they could be expandable as well, no?

Reply:
Yes that is why the for loop for expansion iterates over the entire array

Reply:
My bad, I didn't see that this was already inside `getShellCommand`. Fine then.